encoder-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -0,0 +1,212 @@
+    scopeBuilder.setName(BraveScope.getName());
+    scopeBuilder.setVersion(BraveScope.getVersion());
+    scopeSpanBuilder.setScope(scopeBuilder.build());
+    resourceSpansBuilder.addScopeSpans(scopeSpanBuilder
+        .build());
+    tracesDataBuilder.addResourceSpans(resourceSpansBuilder.build());
+    return tracesDataBuilder.build();
+  }
+
+  static KeyValue stringAttribute(String key, String value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setStringValue(value))
+        .build();
+  }
+
+  static KeyValue intAttribute(String key, int value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setIntValue(value))
+        .build();
+  }
+
+  private Span.Builder builderForSingleSpan(MutableSpan span, Builder resourceSpansBuilder) {
+    Span.Builder spanBuilder = Span.newBuilder()
+        .setTraceId(span.traceId() != null ? ByteString.fromHex(span.traceId()) : INVALID_TRACE_ID)
+        .setSpanId(span.id() != null ? ByteString.fromHex(span.id()) : INVALID_SPAN_ID);
+    if (span.name() != null) {
+      spanBuilder.setName(span.name());
+    }
+    if (span.parentId() != null) {
+      spanBuilder.setParentSpanId(ByteString.fromHex(span.parentId()));
+    }
+    long start = span.startTimestamp();
+    long finish = span.finishTimestamp();
+    spanBuilder.setStartTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(start));
+    spanBuilder.setEndTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(finish));
+    Kind kind = span.kind();
+    if (kind != null) {
+      switch (kind) {
+        case CLIENT:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CLIENT);
+          break;
+        case SERVER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_SERVER);
+          break;
+        case PRODUCER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_PRODUCER);
+          break;
+        case CONSUMER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CONSUMER);
+          break;
+      }
+    }
+    else {
+      spanBuilder.setKind(DEFAULT_KIND);
+    }
+    String localServiceName = span.localServiceName();
+    if (localServiceName == null) {
+      localServiceName = Resource.getDefault().getAttribute(ServiceAttributes.SERVICE_NAME);
+    }
+    resourceSpansBuilder.getResourceBuilder().addAttributes(stringAttribute(ServiceAttributes.SERVICE_NAME.getKey(), localServiceName));
+    maybeAddStringAttribute(spanBuilder, NetworkAttributes.NETWORK_LOCAL_ADDRESS.getKey(), span.localIp());
+    maybeAddIntAttribute(spanBuilder, NetworkAttributes.NETWORK_LOCAL_PORT.getKey(), span.localPort());
+    maybeAddStringAttribute(spanBuilder, NetworkAttributes.NETWORK_PEER_ADDRESS.getKey(), span.remoteIp());
+    maybeAddIntAttribute(spanBuilder, NetworkAttributes.NETWORK_PEER_PORT.getKey(), span.remotePort());
+    maybeAddStringAttribute(spanBuilder, PEER_SERVICE, span.remoteServiceName());
+    span.forEachTag(consumer, spanBuilder);
```

Comment:
I think it's worth naming the field `tagMapper` or `tagToAttribute`. Had to scroll up a lot to find what this consumer does.

Reply:
Fixed in https://github.com/openzipkin-contrib/zipkin-otel/pull/9/commits/d169dfd44ef35abceff88ab03eedbb41536b7a90